common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -2071,6 +2070,154 @@ def copy(self, user_id, source_course, destination_course, subtree_list=None, bl
+
+        IMPORTANT: This method does not preserve block_id - in other words, every block that is
+        copied will be assigned a new block_id. This is because we assume that the same source block
+        may be copied into one course in multiple places. However, it *is* guaranteed that every
+        time this method is called for the same source block and dest_usage, the same resulting
+        block id will be generated.
+
+        :param source_keys: a list of BlockUsageLocators. Order is preserved.
+
+        :param dest_usage: The BlockUsageLocator that will become the parent of an inherited copy
+        of all the xblocks passed in `source_keys`.
+
+        :param user_id: The user who will get credit for making this change.
+        """
+        # Preload the block structures for all source courses/libraries/etc.
+        # so that we can access descendant information quickly
+        source_structures = {}
+        for key in source_keys:
+            course_key = key.course_key.for_version(None)
+            if course_key.branch is None:
+                raise ItemNotFoundError("branch is required for all source keys when using inherit_copy")
+            if course_key not in source_structures:
+                with self.bulk_operations(course_key):
+                    source_structures[course_key] = self._lookup_course(course_key).structure
+
+        destination_course = dest_usage.course_key
+        with self.bulk_operations(destination_course):
+            index_entry = self.get_course_index(destination_course)
+            if index_entry is None:
+                raise ItemNotFoundError(destination_course)
+            dest_structure = self._lookup_course(destination_course).structure
+            old_dest_structure_version = dest_structure['_id']
+            dest_structure = self.version_structure(destination_course, dest_structure, user_id)
+
+            # Set of all descendent block IDs of dest_usage that are to be replaced:
+            block_key = BlockKey(dest_usage.block_type, dest_usage.block_id)
+            orig_descendants = set([block for block in self.descendants(dest_structure['blocks'], block_key, depth=None, descendent_map={})])
+            orig_descendants.remove(block_key)  # The descendants() method used above adds the block itself, which we don't consider a descendant.
+            new_descendants = self._inherit_copy(source_structures, source_keys, dest_structure, block_key, user_id)
+
+            # Update the edit info:
+            dest_info = dest_structure['blocks'][block_key]
+
+            # Update the edit_info:
+            dest_info['edit_info']['previous_version'] = dest_info['edit_info']['update_version']
+            dest_info['edit_info']['update_version'] = old_dest_structure_version
+            dest_info['edit_info']['edited_by'] = user_id
+            dest_info['edit_info']['edited_on'] = datetime.datetime.now(UTC)
```

Comment:
@bradenmacdonald this `some_info['edit_info'][...] = ...` is used quite often, so it might make sense encapsulating it into something like

    def _set_info_items(info, **kwargs):
        for key, value in kwargs.items():
            info['edit_info'][key] = value

And use like

    self._set_info_items(dest_info, 
        previous_version=dest_info['edit_info']['update_version'], 
        update_version=old_dest_structure_version,
        ...
    )